src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1430,14 +1402,14 @@ private bool TryParseSigned(long max, ref TX span, out long result)
             if (!TryParseNonNegative(text, ichMin, ichLim, out val))
             {
                 // Check for acceptable NA forms: ? NaN NA and N/A.
-                result = -max - 1;
+                result = null;
                 return IsStdMissing(ref span);
             }
 
             Contracts.Assert(val >= 0);
             if (val > max)
```

Comment:
>if (val > max) [](start = 12, length = 14)

Previously this method was built on the assumption that the minimum valid value equaled the negative of the maximum valid value. That is no longer the case: the minimum valid value for, say, `sbyte` is `sbyte.MinValue` which is -128, or `sbyte.MaxValue` which is 127. However this code as it stands now will not correctly parse -128 into a valid `sbyte`.

Reply:
That is true, I did notice in Dv classes Min values are used as missing values so legal min value would be MinValue -1 which is equal to MaxValue. I believe we need to fix line 1387 above where we are checking for min bounds for negative numbers.

---
In reply to: [212042421](https://github.com/dotnet/machinelearning/pull/683#discussion_r212042421) [](ancestors = 212042421)